tests/unit_test/llm_test.py

```
@@ -2611,4 +2631,170 @@ def test_load_rerank_embedding_model_hf_cache(self):
+            from kairon.actions.definitions.database import ActionDatabase
+            action = ActionDatabase(bot="test_bot", name="test_db_action")
+            action.suffix = "_faq_embd"
+
+            action.retrieve_config = MagicMock(return_value=(mock_config, mock_bot_settings))
+
+            await action.execute(dispatcher, tracker, {})
+
+            mock_query.assert_called_with(bot="test_bot", collection_name="test_collection")
+
+            assert mock_vector_db.llm.llm_type == "openrouter"
+            assert mock_vector_db.llm.llm_secret == "router_key"
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "llm_secret"

(S105)